### ISSUE TITLE
why the RTMDet training time is so long?

### ISSUE THREAD

**Issue by CarryHJR** [open-mmlab/mmyolo#341]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmyolo/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmyolo.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmyolo).

### 🐞 Describe the bug

according the official training log

yolov5_m-v61_syncbn_fast_8xb16-300e_coco_20220917_204944.log.json  --> 12.7654 h
rtmdet_m_8xb32-300e_coco_20220924_132959.log.json --> 32.8271 h

my cal time script

```

json_log = '/home/user/workplace/python/notebooks/mmyolo/resources/rtmdet_m_8xb32-300e_coco_20220924_132959.log.json'

# json_log = '/home/user/workplace/python/notebooks/mmyolo/resources/yolov5_m-v61_syncbn_fast_8xb16-300e_coco_20220917_204944.log.json'

import json
from collections import defaultdict

log_dict = dict()
with open(json_log, 'r') as log_file:
    for line in log_file:
        log = json.loads(line.strip())
        # skip lines without `epoch` field
        if 'epoch' not in log:
            continue
        epoch = log.pop('epoch')
        if epoch not in log_dict:
            log_dict[epoch] = defaultdict(list)
        for k, v in log.items():
            log_dict[epoch][k].append(v)

all_times = []
include_outliers = False
for epoch in log_dict.keys():
    if include_outliers:
        all_times.append(log_dict[epoch]['time'])
    else:
        all_times.append(log_dict[epoch]['time'][1:])
all_times = np.array(all_times)
print(len(all_times))
np.sum(all_times) * 50 / 3600
print(f'average iter time: {np.mean(all_times):.4f} s/iter')
print(f'total time: {np.sum(all_times) * 50 / 3600:.4f} h')
```

### Environment

no need train

**Comment by RangeKing** [comment]
@CarryHJR, there might be three reasons.
1. (**main reason**) The training of RTMDet didn't use `YOLOv5DetDataPreprocessor` and `yolov5_collate` to preprocess data.
2. RTMDet-m is a little bit larger than YOLOv5-m. 
3. The assigner of RTMDet didn't run in batches.

**Comment by RangeKing** [comment]
@CarryHJR, fast version of RTMDet have released in MMYOLO v0.3.0.

**Comment by CarryHJR** [comment]
for rtmdet_m_syncbn_fast_8xb32-300e_coco_20230102_135952.log.json, we can get total time: 18.1756 h,  great progress !!
